### PR TITLE
Fixed requestAnimationFrame for good

### PR DESCRIPTION
**Pull request by ghost**
requestAnimationFrame was changed recently. The new version is used in
Chrome 24 and IE 10. They had changed the first argument. Instead of
sending the time stamp starting from epoch, they now send more precise
timestamp in miliseconds starting from page load. So to resolve
animation start window.perfomance.now should be used.

The discussion occured in mailing lists back in May:
http://lists.w3.org/Archives/Public/public-web-perf/2012May/thread.html#msg36

### CODE REVIEW

File: src/morpheus.js [hunk 1]
```diff
@@ -172,12 +167,13 @@
       , self = this
       , diff = to - from
       , start = +new Date()
+      , hrStart = win.performance && win.performance.now && win.performance.now()
```
Comment (ded) [inline src/morpheus.js]: can you move this detection outside the frame method? so it doesn't check for `win.performance.now` upon every frame request.

**Comment by ded** [comment]
currently not merging...

File: src/morpheus.js (hunk 1 above)
Comment (ghost) [inline src/morpheus.js]: nope :-( this code throws "TypeError: Illegal invocation" in chrome:
`var a = window.performance.now; a();`

also i've put the msRequestAnimationFrame back, and made one more slight check, just to be sure.

Comment (ded) [inline src/morpheus.js]: it should work... what i meant is this...

``` js
// somewhere outside
var hasPerf = win.performance && win.performance.now


 // then inside the frame method...
var time = duration || thousand
      , self = this
      , diff = to - from
      , start = +new Date()
      , stop = 0
      , end = 0
    live(run)

    function run(t) {
      var delta = (t < 1e12 && hasPerf) ? t - win.performance.now() : t - start
      if (delta > time || stop) {
        to = isFinite(to) ? to : 1
        stop ? end && fn(to) : fn(to)
        die(run)
        return done && done.apply(self)
      }

// ...
```

**Comment by ghost** [comment]
done

**Comment by ded** [comment]
this still isn't merging. can you clean this up for a clean merge?

**Comment by ghost** [comment]
no. i mean do whatever you feel is necessary.

**Comment by ded** [comment]
is still breaking?

**Comment by danro** [comment]
@ded After review, it looks like this will conflict because it's a different approach to dealing with the new requestAnimationFrame timers. The currently merged approach #34, simply sidesteps the high-res timer features offered in IE10 and Chrome 24, where as this pull requests attempts to add support.

I'd like to see the high-res timer supported, myself- rather than force the new browsers to use the older (low-res) timer methods. Here's an article that covers the new timer changes:

http://updates.html5rocks.com/2012/05/requestAnimationFrame-API-now-with-sub-millisecond-precision

**Comment by ded** [comment]
ugh. ok. this explains a lot. however we should be using `performance.now` — this is what i get for not full reading the new api

**Comment by danro** [comment]
Here's an interesting snippet.

`startTime = window.performance && window.performance.now ? (performance.now() + performance.timing.navigationStart) : +new Date();`

Can't we just change the start time? Seems like it would be more performant than checking the time value on each `run()`

**Pull request closed without merging**